pkg/apis/metering/v1/metering_config.go

```diff
@@ -0,0 +1,489 @@
+	Bucket          string `json:"bucket,omitempty"`
+	CreateSecret    bool   `json:"createSecret"`
+	Endpoint        string `json:"endpoint,omitempty"`
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+	SecretName      string `json:"secretName,omitempty"`
+}
+
+type SharedPVCSpec struct {
+	ClaimName    string `json:"claimName,omitempty"`
+	CreatePVC    bool   `json:"createPVC,omitempty"`
+	MountPath    string `json:"mountPath,omitempty"`
+	Size         string `json:"size,omitempty"`
+	StorageClass string `json:"storageClass,omitempty"`
+}
+
+/*
+End of storage section
+*/
+
+type PrestoLabels struct {
+}
+type PrestoImage struct {
+	PullPolicy string `json:"pullPolicy,omitempty"`
+	Repository string `json:"repository,omitempty"`
+	Tag        string `json:"tag,omitempty"`
+}
+type PrestoAWS struct {
+	AccessKeyID     string `json:"accessKeyID,omitempty"`
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+	CreateSecret    bool   `json:"createSecret"`
+	SecretName      string `json:"secretName,omitempty"`
+}
+type PrestoAzure struct {
+	StorageAccountName string `json:"storageAccountName,omitempty"`
+	SecretAccessKey    string `json:"secretAccessKey,omitempty"`
+	CreateSecret       bool   `json:"createSecret"`
+	SecretName         string `json:"secretName,omitempty"`
+}
+type PrestoGcs struct {
+	CreateSecret          bool   `json:"createSecret"`
+	SecretName            string `json:"secretName,omitempty"`
+	ServiceAccountKeyJSON string `json:"serviceAccountKeyJSON,omitempty"`
+}
+type PrestoS3Compatible struct {
+	SecretName      string `json:"secretName,omitempty"`
+	CreateSecret    string `json:"createSecret"`
+	AccessKeyID     string `json:"accessKeyID,omitempty"`
+	SecretAccessKey string `json:"secretAccessKey,omitempty"`
+	Endpoint        string `json:"endpoint,omitempty"`
+}
+type PrestoTLS struct {
+	Certificate   string `json:"certificate,omitempty"`
+	Key           string `json:"key,omitempty"`
+	CaCertificate string `json:"caCertificate,omitempty"`
+	CreateSecret  bool   `json:"createSecret"`
+	Enabled       bool   `json:"enabled"`
+	SecretName    string `json:"secretName,omitempty"`
+}
+type PrestoAuth struct {
+	Certificate   string `json:"certificate,omitempty"`
+	Key           string `json:"key,omitempty"`
+	CaCertificate string `json:"caCertificate,omitempty"`
+	CreateSecret  bool   `json:"createSecret"`
+	Enabled       bool   `json:"enabled"`
+	SecretName    string `json:"secretName,omitempty"`
+}
+type PrestoHive struct {
+	HadoopConfigSecretName string `json:"hadoopConfigSecretName,omitempty"`
+	UseHadoopConfig        bool   `json:"useHadoopConfig,omitempty"`
+	MetastoreURI           string `json:"metastoreURI,omitempty"`
+	MetastoreTimeout       string `json:"metastoreTimeout,omitempty"`
+	TLS                    *TLS   `json:"tls,omitempty"`
+}
+type Connectors struct {
+	Hive PrestoHive `json:"hive,omitempty"`
+	// ExtraConnectorFiles []interface{} `json:"extraConnectorFiles,omitempty"`
```

Comment:
the type for this is (uses an anonymous struct):
```
type ConnectorFileList []struct {
   Name string `json:"name"`
   Content string `json:"content"`
}
```

Reply:
It seems like this needs to be a defined structure, as `make update-codegen` appears to fail anytime I define a (nested) anonymous structure.

Reply:
That's fine.